format-specs/compatible-parquet.md

```diff
@@ -0,0 +1,41 @@
+
+To be clear, this is *only* recommended for those who are using tools that don't yet produce valid GeoParquet, and we encourage advocating to the creaters of tools you are using to implement the GeoParquet spec. Feel free to [start a discussion](https://github.com/opengeospatial/geoparquet/discussions) to raise awareness of tools that ideally support Geoparquet - the community can likely help in encouraging an implementation.
+
+The core idea behind these compatibility guidelines is that tools and libraries that read GeoParquet will be able to parse these geospatial compatible Parquet files, to make it easy to get data into the GeoParquet ecosystem. But it is only recommended for those tools and libraries to produce valid GeoParquet, following [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle) of being liberal in what you accept but conservative in what you send - if you are authoring a tool to write GeoParquet please do not give users the option to create these parquet files.
+
+## Compatibility Guidelines
+
+The core idea of the compatibility guidelines is to have the output match the defaults of the official spec as closely as possible, so it is very easy for tools to simply add the appropriate Parquet metadata and create valid GeoParquet. The guidelines are as follows:
+
+* The geometry column should be named 'geometry'.
+
+* The geometry column should be a `BYTE_ARRAY` with Well Known Binary (WKB) used to define the geometries, as defined in the [encoding](./geoparquet.md#encoding) section of the GeoParquet spec.
+
+* All data is stored in longitude, latitude based on the WGS84 datum, as defined as the default in the [crs](./geoparquet.md#crs) section of the GeoParquet spec.
+
+### Data Reader Assumptions
+
+The above are the key recommendations a data producer should follow. Any implemented reader will need to make the following assumptions when reading one of these columns, unless the user supplies additional information that they are aware of:
+
+* The geometry_types values is an empty array, signaling the geometry type is not known and the reader should make no assumptions about the types, as defined in the [geometry_types](./geoparquet.md#geometry_types) section of the spec.
+
+* Any CRS-aware reader should assume that the CRS is OGC:CRS84 as explained in the [crs](./geoparquet.md#crs) section of the spec. (Or it could assume it is EPSG:4326 but overriding the axis order to assume longitude latitude as explained in the [Coordinate axis order](./geoparquet.md#coordinate-axis-order)) section.
+
+* No assertions are made on the winding order, the default of the [orientation](./geoparquet.md#orientation) section of the spec.
+
+* Edges are planar, as explained to be the default in the [edges](./geoparquet.md#edges) section of the spec.
```

Comment:
I am wondering how practical this requirement / assumption is. 
Assume you are using BigQuery, and you have a geography column, and you then export the table to Parquet (https://cloud.google.com/bigquery/docs/exporting-data#parquet_export_details). The docs don't mention how it will represent a geography column, but I assume it will be WKT or WKB. And I also assume that it will then use spherical edges, since this is what Bigquery assumes when loading data from WKT/WKB (https://cloud.google.com/bigquery/docs/reference/standard-sql/geography_functions#st_geogfromtext). And I don't directly see a way to manually create a new column of binary/string type with the WKB/WKT values that have planar edges (the `ST_ASTEXT` or `ST_ASBINARY` don't provide any options for it)

(of course, we could say this will require the user to specify the edges through some option when reading this kind of data. But if you didn't produce the data yourself, you might not necessarily know this)

Reply:
I agree it's not a great assumption, but I think we have to assume something? Unless there's some way to automatically detect this? 

Hmmm, just had an idea that might be bad, but we could say that if the column is named 'geography' then we assume edges are spherical? That little bit of naming of the column seems to be the only place where we can communicate a bit of metadata.

But I think we should have a clear target for this - a product that does need to export where this helps. For BigQuery we already have https://github.com/geoparquet/bigquery-converter - and the readme there makes it clear that when you try to export a geography column to Parquet it just says `Error while reading data, error message: Type GEOGRAPHY is not currently supported for parquet exports.`

I also think it's ok to have it be something for users to supply. I also think we shouldn't go too far in bending over backwards for these compatible things - we more push for tools to implement. It also seems like people could use GeoPandas pretty easily to take most any Parquet geospatial output and get it in the right shape to write out to GeoParquet.

Reply:
Yes, BigQuery throws an error when trying to export a parquet with a geography column, and as @jorisvandenbossche pointed out it's not possible to export in planar edges.

I have to say I'm not a great fan of using `geography` or `geometry` to define the edges, but I don't have a better solution, due to the importance of data warehouses in this initiative, and because Snowflake and BigQuery only support in production geography data types, I will go with the solution @cholmes proposed

The export command from BigQuery would be something like this:

```sql
EXPORT DATA
  OPTIONS (
    uri = 'gs://alasarr/geoparquet/*.parquet',
    format = 'parquet')
AS (
select ST_ASBINARY(geom) as geography FROM ...
)
```

Reply:
Cool. That reminds me one other thing that we discussed on the call that pointed towards having geometry as the default. I'm not sure if I've got it right, and @paleolimbot made the point, so he can correct me. But basically in his experience working on an s2 library it was more likely for a line defined by just two points to be a geometry - that lines with geography edges were more likely to include a number of interpolated points along the line.

Reply:
Yes, BigQuery works like that. It adds additional points to convert to geodesic: https://cloud.google.com/bigquery/docs/reference/standard-sql/geography_functions#st_geogfromgeojson

Reply:
Do we want to add that "geometry" vs "geography" column name and default for planar vs spherical edges to the document?

Reply:
I think we should. @cholmes do you want me to create a PR for this?

Reply:
PR up at https://github.com/opengeospatial/geoparquet/pull/175